### PR TITLE
Add Markdown File Viewer

### LINKED ISSUE #1365: Markdown File Viewer
### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [X] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

The app currently uses the browser's default viewers for PDF and PNG files, lacking support for viewing Markdown files. 
The desired behavior is to augment the app by integrating a Markdown viewer, enabling users to view Markdown files within the application.

### Mention any other details that might be useful
* I have already implemented a Markdown viewer as part of a project that used this app as its foundation.
The viewer is based on this library: [react-markdown-preview](https://www.npmjs.com/package/@uiw/react-markdown-preview)
Please let me know if you would like me to open a PR to add this Markdown viewer.

* Currently Markdown files are not processed in this app and there is no markdown file processor, I can add that as well (perhaps as a part of a new issue?)

gif

### PR DESCRIPTION
**Pull request by yuvalyaron**
## Purpose

The app currently uses the browser's default viewers for PDF and PNG files, lacking support for viewing Markdown files.
The desired behavior is to augment the app by integrating a Markdown viewer, enabling users to view Markdown files within the application.

Depends on #1381 
Resolves #1365 

image

gif

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

Tested manually only.

- [ ] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works
- [ ] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ ] I ran `python -m mypy` to check for type errors
- [X] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

File: app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx [hunk 1]
```diff
@@ -29,30 +30,46 @@ export const AnalysisPanel = ({ answer, activeTab, activeCitation, citationHeigh
...
-            const response = await fetch(activeCitation, {
-                method: "GET",
-                headers: getHeaders(token)
-            });
-            const citationContent = await response.blob();
-            let citationObjectUrl = URL.createObjectURL(citationContent);
-            // Add hash back to the new blob URL
-            if (originalHash) {
-                citationObjectUrl += "#" + originalHash;
+        const fetchCitation = async () => {
```
Comment (pamelafox) [inline app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx:35]: Did all these lines need to change?

Comment (pamelafox) [reply app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx:35]: Ah I see this PR is in draft mode so you may still be working on it.

Comment (yuvalyaron) [reply app/frontend/src/components/AnalysisPanel/AnalysisPanel.tsx]: That was a mistake - fixed that, thanks!
The PR is ready to be reviewed, but it's waiting for #1381 so it's still in draft

**Comment by pamelafox** [comment]
Hi @yuvalyaron - Sorry for the delay, I wanted to research the package you're proposing using, uiw/react-markdown, versus the package used by the -javascript repo, marked.

Unfortunately, uiw/react-markdown doesn't look like a good fit for integrating in a big generic repository like this since:

1) it has 12 dependencies (versus 0 for marked)
2) it has a publish size of 4 MB and download size of 10 MB (versus 900K for marked)
3) it has a lower score on snyk.io (81 vs 100 for marked)

So I think we should use the marked package instead.

**Comment by yuvalyaron** [comment]
Hi @pamelafox, thanks for the review!
I've switched the code to use marked instead. Though the visual outcome may not be as impressive, I agree that marked is a more suitable choice for the repository based on the considerations you've outlined.


image

File: app/frontend/src/components/MarkdownViewer/MarkdownViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+interface MarkdownViewerProps {
+    src: string;
+}
+
+export const MarkdownViewer: React.FC<MarkdownViewerProps> = ({ src }) => {
+    const [content, setContent] = useState<string>("");
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+    const [error, setError] = useState<Error | null>(null);
+
+    const removeAnchorLinks = (html: string) => {
```
Comment (pamelafox) [inline app/frontend/src/components/MarkdownViewer/MarkdownViewer.tsx:15]: Can you comment as to why this is needed?

File: app/frontend/src/components/MarkdownViewer/MarkdownViewer.module.css [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+tr:nth-child(even) {
+    background-color: #f6f8fa;
+}
+
+code {
+    display: block;
+    font-family: monospace;
+    padding: 10px;
+    background-color: #f6f8fa;
+}
```
Comment (pamelafox) [inline app/frontend/src/components/MarkdownViewer/MarkdownViewer.module.css:50]: Did you run the pre-commit? It usually fixes EOF newlines.

**Comment by pamelafox** [comment]
I tried it out and it looks nice! I just had a few comments/questions.

File: app/frontend/src/components/MarkdownViewer/MarkdownViewer.tsx (hunk 2 above)
Comment (yuvalyaron) [reply app/frontend/src/components/MarkdownViewer/MarkdownViewer.tsx:19]: Added

File: app/frontend/src/components/MarkdownViewer/MarkdownViewer.module.css (hunk 3 above)
Comment (yuvalyaron) [reply app/frontend/src/components/MarkdownViewer/MarkdownViewer.module.css]: 👍 Ran now

**Review by pamelafox — APPROVED** [review]
Looks good, I'll plan to merge on Monday. Thank you!

**Pull request merged**